refinery/tool_manager/models.py

```diff
@@ -255,12 +259,16 @@ def launch(self):
                         self.file_relationships
                     )
                 },
-                manager=get_django_docker_engine_manager()
+                extra_directories=ast.literal_eval(
+                    self.tool_definition.extra_directories
+                )
             )
             try:
-                container.run()
+                DockerClientWrapper().run(container)
             except APIError as e:
                 return HttpResponseServerError(content=e)
+            except AssertionError as e:
+                return HttpResponseBadRequest(content=e)
```

Comment:
I like assertions... but I'm not comfortable catching them: Over time, how do you know that it is the assertion you had in mind that's causing the trouble now? Perhaps better to have a new error class, if it's really necessary?